os/include/tinyara/wifi/rtk/wifi_constants.h

```diff
@@ -633,10 +633,12 @@ enum _WIFI_EVENT_INDICATE{
 	WIFI_EVENT_IP_CHANGED = 16,
 	WIFI_EVENT_ICV_ERROR = 17,
 	WIFI_EVENT_CHALLENGE_FAIL = 18,
-	
+
+#ifdef CONFIG_ARCH_CHIP_RTL8720E
 		/* csi rx done event */
-	WIFI_EVENT_CSI_DONE = 160,
-	
+	WIFI_EVENT_CSI_DONE = 15,

```

Comment:
15 is duplicated. Is it ok?

Reply:
Dear @sunghan-chang,

Yes, as RTL8720E does not have WIFI_EVENT_BEACON_AFTER_DHCP  event anymore, and tp1x don't have WIFICSI. Will it be better if i change it to this 

#ifdef CONFIG_ARCH_CHIP_RTL8720E
  		/* csi rx done event */
  	WIFI_EVENT_CSI_DONE = 15,
#else
	WIFI_EVENT_BEACON_AFTER_DHCP = 15,
#endif

Reply:
Suggestion is better I think. And please leave your explanation there.

Reply:
One question, if some other chip needs both, how will you do? No need to concern this? Is there any problem with previous value 160? or other no-duplicated value?

Reply:
Changing it from 160 to 15 is for codesize cost down, and also synchronization with our internal sdk. Choosing other no-duplicated value might cause conflicts in the future when there are more events added in. Do you have any better suggestions for this?

Reply:
> Dear @sunghan-chang,
> 
> Yes, as RTL8720E does not have WIFI_EVENT_BEACON_AFTER_DHCP event anymore, and tp1x don't have WIFICSI. Will it be better if i change it to this
> 
> #ifdef CONFIG_ARCH_CHIP_RTL8720E /* csi rx done event */ WIFI_EVENT_CSI_DONE = 15, #else WIFI_EVENT_BEACON_AFTER_DHCP = 15, #endif

In my opinion, even if it working exclusively, you should define different value.
Please assume that there's framework for both of tp1x & tp1e, if you define same value, then framework also should have a kind of configuration..below is more good approach
```
1. framework
void do_something_with_csi()
{
    ret = send_message(WIFI_EVENT_CSI_DONE);
    if (ret != -ENOSYS) { //board doesn't provide it
           return ret;
    }
    do_something;
}
void do_something_with_dhcp()
{
    ret = send_message(WIFI_EVENT_BEACON_AFTER_DHCP);
    if (ret != -ENOSYS) { //board doesn't provide it
           return ret;
    }
    do_something;
}
2. BSP
void send_message(WIFI_EVENT_INDICATE event)
{
#ifdef CONFIG_ARCH_CHIP_RTL8720E
     if (event == WIFI_EVENT_BEACON_AFTER_DHCP) {
          return -ENOSYS;
     }
#endif

#ifdef CONFIG_ARCH_CHIP_RTL8721
    if (event == WIFI_EVENT_CSI_DONE) {
          return -ENOSYS;
    }
#endif
   do_something;
}

3.enum
enum _WIFI_EVENT_INDICATE {
	WIFI_EVENT_BEACON_AFTER_DHCP = 15,
         ....
        WIFI_EVENT_CSI_DONE = 19,
         ....
}